vault/logical_system.go

```diff
@@ -883,10 +881,11 @@ func (b *SystemBackend) handlePluginCatalogRead(req *logical.Request, d *framewo
 		return nil, nil
 	}
 
+	// Create a map of data to be returned and remove sensitive information from it
+	data := structs.New(plugin).Map()
+
 	return &logical.Response{
-		Data: map[string]interface{}{
-			"plugin": plugin,
-		},
+		Data: data,
```

Comment:
@briankassouf I changed the `logical.Response` on the ReadOperation from the plugin catalog to return the key-value pairs on plugin properties directly instead of having those embedded in a map. There should be no change needed on the plugin implementations themselves. The tests have also been updated to pass with this new response object.

Before:
```
$  vault read /sys/plugins/catalog/mysql-database-plugin
Key   	Value
---   	-----
plugin	map[name:mysql-database-plugin command: args:<nil> sha256:<nil> builtin:true]
```

After:
```
$ vault read /sys/plugins/catalog/mysql-database-plugin
Key    	Value
---    	-----
args   	<nil>
builtin	true
command
name   	mysql-database-plugin
sha256 	<nil>
```